src/BoundaryConditions.h

```diff
@@ -105,21 +93,23 @@ inline NO_INLINE Func constant_exterior(T func_like, Tuple value) {
     return constant_exterior(Internal::func_like_to_func(func_like), value, object_bounds);
 }
 template <typename T>
-inline NO_INLINE Func constant_exterior(T func_like, Expr value) {
+inline NO_INLINE Func constant_exterior(const T &func_like, Expr value) {
     return constant_exterior(func_like, Tuple(value));
 }
 
-template <typename T, typename ...Bounds>
-inline NO_INLINE Func constant_exterior(T func_like, Tuple value,
-                                        Bounds... bounds) {
+template <typename T, typename ...Bounds,
+          typename std::enable_if<Halide::Internal::all_are_convertible<Expr, Bounds...>::value>::type* = nullptr>
+inline NO_INLINE Func constant_exterior(const T &func_like, Tuple value,
```

Comment:
T could be passed via universal reference as well, so that rvalue Funcs can be given.